drivers/display/display.c

```diff
@@ -0,0 +1,193 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "display.h"
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+
+/***************************************************************************//**
+ * @brief Initializes the display peripheral.
+ *
+ * @param device     - The device structure.
+ * @param param      - The structure that contains the device initial
+ * 		               parameters.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+*******************************************************************************/
+int32_t display_init(struct display_dev **device,
+		     const struct display_init_param *param)
+{
+	struct display_dev *dev;
+	uint32_t ret;
+
+	dev = (struct display_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+	ret = spi_init(&dev->spi_desc, param->spi_ip);
+	if (ret != SUCCESS) {
+		free(dev);
+		return FAILURE;
+	}
+	dev->cols_nb = param->cols_nb;
+	dev->rows_nb = param->rows_nb;
+	dev->controller_ops = param->controller_ops;
+	dev->extra = param->extra;
+
+	ret = dev->controller_ops->init(dev);
+	if (ret != SUCCESS) {
+		free(dev);
+		return FAILURE;
+	}
+
+	*device = dev;
+
+	return ret;
+}
+
+/****************************************************************************//**
+ * @brief Frees the resources allocated by display_init().
+ *
+ * @param device - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+*******************************************************************************/
+int32_t display_remove(struct display_dev *device)
+{
+	int32_t ret;
+
+	ret = device->controller_ops->remove(device);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = spi_remove(device->spi_desc);
+	if (ret != SUCCESS)
+		return FAILURE;
+	free(device);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Turns display on/off.
+ *
+ * @param device - The device structure.
+ * @param on_off - TRUE = on
```

Comment:
Where does the capital `TRUE` and `FALSE` are coming from ? which header ? I think the user should not care.

To simplify the user API, perhaps it would be better to have 2 functions in display.h/.c:
`display_on(dev)`
`display_off(dev)`

These in turn will call the corresponding op with true/false from <stdbool.h>

`display_on_off(dev, true);`
`display_on_off(dev, false);`